File: HasCycle.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public bool HasCycle(ListNode head) {
        
        if (head == null)
        {
            return false;
        }
        
        List<ListNode> NodeList = new List<ListNode>();
        
        while (head != null)
        {
            if (NodeList.Contains(head))
            {
                return true;
            }
            
            NodeList.Add(head);
            head = head.next;
        }
        
        return false;
    }
}

File: Python/hasCycle.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:

        seenVals = []

        while head:
            if head in seenVals:
                return True

            seenVals.append(head)
            head = head.next

        return False
